### DISCUSSION TITLE
Sending additional request to server from proxy

### DISCUSSION THREAD

**Discussion by sshantha22** [abhinavsingh/proxy.py#1392] [General]
Hi,
I have a use case where when the proxy receives a request from the client, the proxy will need to send its own request to the server to setup the context in the server before sending the client request to the server. Is there an example as to how I can do this?
Thanks

**Comment by abhinavsingh** [comment]
Hi @sshantha22 , you can try [`before_upstream_connection`](https://github.com/abhinavsingh/proxy.py/blob/develop/proxy/http/proxy/plugin.py#L75) .  Within this callback, make your own request to the server and only then return `request` at the end of this function.   By doing so, your proxy requests will not get processed, until you return from this callback.

**Reply by sshantha22** [reply]
Thank you Abhinav Singh. I might need to make periodic requests once the connection has been established with the server within the context of a particular session and not as separate requests outside of the session. If I make the call in before_upstream_connection, would it be outside of the session? I was thinking of making the proxy initiated request in handle_client_request. Please let me know your thoughts. If you have an example, that would be very helpful.
